packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -359,7 +354,7 @@ function computeShadowMode(def: ComponentDef, owner: VM | null) {
                     // transitively opts all of its descendants into native.
                     // Synthetic if neither this component nor any of its ancestors are configured
                     // to be native.
```

Comment:
This comment doesn't sound correct anymore. In this case, the shadowMode is inherited from its parent and falls back to SyntheticShadow if it's the root component.

```suggestion
                    // Synthetic if it is the root component.
```